MessageHandler.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <binaryninjaapi.h>
+
+class MessageHandler {
+
```

Comment:
I feel like there is a better name for this class; `FunctionCache` comes to mind, as that's a more accurate description of what it is. I just don't like `MessageHandler` on the basis that this class does not handle messages.

Reply:
It will later handle message metadata for objc stubs and likely other stuff in the future. (see https://github.com/Vector35/workflow_objc/commit/8209c2d915d217332f8ae5b07a78bd31aa29e0f1#diff-c054eb033c6e47748eaea2e9d711e5c76224e7f0a67cdc0d6c897a430ed2be57 , this is a derivative of that. )  

"MessageHandler" was the most appropriate name I could think of given the larger responsibilities it will need to have later down the line.

Reply:
Oh, I must have missed that in the documentation comment explaining that class... 😛 

Serious reply: Makes sense and sounds good.